bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java

```diff
@@ -403,6 +439,7 @@ public void handleUpdate(FeatureDataDTO featureDataDTO) {
                                 case "percent":
                                 case "minute":
                                 case "hours":
+                                case "kelvin":
                                     updateState(msg.getChannelId(),
                                             new QuantityType<>(Double.valueOf(msg.getValue()) + " " + unit));
```

Comment:
Didn't notice that before: Isn't `msg.getValue()` returning a string that can be used without converting it to a Double before? This looks like you are converting a String to a Double and back to a String.